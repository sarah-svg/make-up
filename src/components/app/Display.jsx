import React from 'react';
import PropTypes from 'prop-types';
import './styles.css';

const Display = ({ red, amber, green, onClick }) => (
  <div className="black-box" onClick={onClick}>
    <div className={`light ${red ? 'red' : 'inactive'}`} />
    <div className={`light ${amber ? 'amber' : 'inactive'}`} />
    <div className={`light ${green ? 'green' : 'inactive'}`} />
  </div>

);


Display.propTypes = {
  red: PropTypes.bool.isRequired,
  amber: PropTypes.bool.isRequired,
  green: PropTypes.bool.isRequired,
  onClick: PropTypes.func.isRequired,
};

export default Display;
