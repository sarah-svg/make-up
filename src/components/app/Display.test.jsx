

import React from 'react';
import { render, cleanup } from '@testing-library/react';
import Display from './Display';

describe('App component', () => {
  afterEach(() => cleanup());
  it('renders App', () => {
    const { asFragment } = render(<Display />);
    expect(asFragment()).toMatchSnapshot();
  });
});
