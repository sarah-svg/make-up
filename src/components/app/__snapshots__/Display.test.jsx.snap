// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App component renders App 1`] = `
<DocumentFragment>
  <div
    class="black-box"
  >
    <div
      class="light inactive"
    />
    <div
      class="light inactive"
    />
    <div
      class="light inactive"
    />
  </div>
</DocumentFragment>
`;
